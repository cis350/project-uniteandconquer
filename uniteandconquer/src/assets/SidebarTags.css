.preferences {
  position: absolute;
  width: 302px;
  height: 300px;
  left: 51px;
  top: 461px;

  background: #FFD9A0;

}

.preferences-box {
  background: #FFD9A0;
  padding: 20px;
}

.notifications {
  position: absolute;
  width: 302px;
  left: 51px;
  top: 800px;

  background: #FFD9A0;
}

.preferences-title {
  font-size: 25px;
  font-weight: bold;
  font-style: italic;
  text-align: center;
  padding: 10px;
}

.preference-tag {
  background: #FFAB2D;
  border: 5px solid #FFFFFF;
  box-sizing: border-box;
  border-radius: 40px;
  text-align: center;
  width: 150px;
  height: 50px;
  margin: auto;
  margin-bottom: 20px;
  padding-top: 9px;
}

.settings-box:hover {
  background-color: #ffd9a0;
}

.settings-box {
  justify-content: center;
  align-items: center;
  border: 16px solid #FFD9A0;
  padding-bottom: 5px;
  padding-top: 10px;
  margin-bottom: 40px;
}

.settings {
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  padding: 10px;

  position: absolute;
  width: 303px;
  height: 98px;
  left: 50px;
  top: 324px;

  background: #FAE9C7;
  border: 16px solid #FFD9A0;
  box-sizing: border-box;
  box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
}

.home-box:hover {
  background-color: #ffd9a0;
}

.home-box {
  justify-content: center;
  align-items: center;
  border: 16px solid #FFD9A0;
  padding-bottom: 5px;
  padding-top: 10px;
  margin-bottom: 40px;
}

.home {
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  padding: 10px;

  position: absolute;
  width: 303px;
  height: 98px;
  left: 51px;
  top: 187px;

  background: #FAE9C7;
  border: 16px solid #FFD9A0;
  box-sizing: border-box;
  box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
}