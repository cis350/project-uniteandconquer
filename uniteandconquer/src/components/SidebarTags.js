import React, { useRef } from 'react';
import { Link } from 'react-router-dom';
import '../assets/SidebarTags.css';

function SidebarTags() {
  const tags = useRef([
    'Appliances', 'Beauty & Care', 'Health & Household', 'Electronics',
  ]);
  return (
    <div className="sidebar">
      <div className="greeting-box">
        <i className="far fa-user-circle fa-2x" />
        {' '}
        Hi, Jeremy
      </div>

      <div className="home-box">
        <Link className="link" to="/">
          <div className="text">Home</div>
        </Link>
      </div>
      <div className="settings-box">
        <Link className="link" to="/user-settings-personal-information">
          <div className="text">Settings</div>
        </Link>
      </div>
      <div className="preferences-box">
        <div className="preferences-title">My Preferences</div>
        {tags.current.map((tag) => (
          <button type="button" key={tag} className="preference-tag">{tag}</button>
        ))}
      </div>
    </div>
  );
}

export default SidebarTags;
