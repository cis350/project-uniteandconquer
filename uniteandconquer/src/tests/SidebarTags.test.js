/**
 * @jest-environment jsdom
 */

import React from 'react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import {
  render, screen,
} from '@testing-library/react';
import SidebarTags from '../components/SidebarTags';

// https://stackoverflow.com/questions/66284286/react-jest-mock-usenavigate

const mockedUsedLink = jest.fn();
const mockedUsedNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  Link: () => mockedUsedLink,
  useNavigate: () => mockedUsedNavigate,
}));

test('sidebar header (main/home page) content', () => {
  render(<SidebarTags />);
  const headerElement = screen.getByTestId('home-header');
  expect(headerElement).toBeInTheDocument();
});

test('sidebar header (settings) content', () => {
  render(<SidebarTags />);
  const headerElement = screen.getByTestId('settings-header');
  expect(headerElement).toBeInTheDocument();
});

test('sidebar header (preferences) content', () => {
  render(<SidebarTags />);
  const headerElement = screen.getByTestId('preferences-header');
  expect(headerElement).toBeInTheDocument();
});

test('snapshot sidebar for tags', () => {
  const component = renderer.create(<SidebarTags />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
