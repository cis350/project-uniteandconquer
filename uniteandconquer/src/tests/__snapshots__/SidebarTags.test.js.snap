// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot sidebar for tags 1`] = `
<div
  className="sidebar"
>
  <div
    className="greeting-box"
  >
    <i
      className="far fa-user-circle fa-2x"
    />
     
    Hi, Jeremy
  </div>
  <div
    className="home-box"
    data-testid="home-header"
  />
  <div
    className="settings-box"
    data-testid="settings-header"
  />
  <div
    className="preferences-box"
    data-testid="preferences-header"
  >
    <div
      className="preferences-title"
    >
      My Preferences
    </div>
    <button
      className="preference-tag"
      type="button"
    >
      Appliances
    </button>
    <button
      className="preference-tag"
      type="button"
    >
      Beauty & Care
    </button>
    <button
      className="preference-tag"
      type="button"
    >
      Health & Household
    </button>
    <button
      className="preference-tag"
      type="button"
    >
      Electronics
    </button>
  </div>
</div>
`;
